### PR TITLE
Fix file includes

### PR DESCRIPTION
**Pull request by Malax**
Cargo cannot include files outside of a package's root directory. We currently have includes that violate this rule. This never surfaced since we don't depend on these files being included and Cargo doesn't warn when a file couldn't be included. This isn't well documented either. (refs: https://doc.rust-lang.org/cargo/reference/manifest.html#the-exclude-and-include-fields, https://github.com/rust-lang/cargo/issues/10617#issuecomment-1114361442)

The workaround here is using symlinks which will be packaged up correctly. I tested this via [`cargo package`](https://doc.rust-lang.org/cargo/commands/cargo-package.html), both using `--list` as well as inspecting the resulting `.crate` file.

In addition, the README file for libcnb is located outside of the package root directory as well. Since we include that file as documentation in the source, compilation fails when the packaged crate is being compiled from scratch. This prevents vendoring (via [`cargo vendor`](https://doc.rust-lang.org/cargo/commands/cargo-vendor.html)) of `libcnb`.

I'm not entirely sure why the README include doesn't work with `cargo vendor` since the `README.md` is correctly included in the `.crate` file I manually downloaded from crates.io. I suspect that the Cargo metadata that references the README file outside of the package root directory makes `cargo vendor` skip this file. This also means I cannot 💯 confirm this PR fixes the vendoring issue without publishing to crates.io.

Closes GUS-W-11828824

### CODE REVIEW

**Review by edmorley — APPROVED** [review]
Thank you!

Sounds like this would be a useful thing to report upstream too :-) (The lack of a warning)

File: libherokubuildpack/Cargo.toml [hunk 1]
```diff
@@ -8,8 +8,8 @@ description = "Opinionated common code for buildpacks implemented with libcnb.rs
 keywords = ["buildpacks", "CNB"]
 repository = "https://github.com/heroku/libcnb.rs/tree/main/libherokubuildpack"
 documentation = "https://docs.rs/libherokubuildpack"
-readme = "./README.md"
-include = ["src/**/*", "../LICENSE", "../README.md"]
+readme = "README.md"
+include = ["src/**/*", "../LICENSE", "README.md"]
```
[the remark this answers is not in the document]
Comment (Malax) [reply libherokubuildpack/Cargo.toml:12]: The linked Cargo issue talks about adding warnings. :)

**Comment by edmorley** [comment]
> Sounds like this would be a useful thing to report upstream too :-) (The lack of a warning)

Ah there is already: https://github.com/rust-lang/cargo/issues/9667

**Comment by edmorley** [comment]
Also: https://github.com/rust-lang/cargo/issues/5911

**Comment by Malax** [comment]
Version `0.11.1` has been released and the vendoring issue is indeed fixed now. 🎉

**Pull request merged**